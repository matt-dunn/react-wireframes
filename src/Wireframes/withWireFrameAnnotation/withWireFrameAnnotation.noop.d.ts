/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */
/// <reference types="react" />
/// <reference types="@emotion/core" />
export declare const withWireFrameAnnotation: <P>(WrappedComponent: string | import("react").ComponentClass<P, any> | import("react").FunctionComponent<P>) => string | import("react").ComponentClass<P, any> | import("react").FunctionComponent<P>;
