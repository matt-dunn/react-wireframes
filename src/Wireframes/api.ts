/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import { ComponentType, ReactNode } from "react";

import { getWireframeAnnotation, updateWireframeAnnotation } from "./utils";

type APIOptions = {
  updater: (annotations: WireframeAnnotations) => void;
  highlightNote?: (annotation?: WireframeAnnotation) => void;
}

export type WireframeAnnotationOptions = {
  title: ReactNode;
  description: ReactNode;
}

export type WireframeAnnotation = {
  id: number;
  Component: ComponentType<any>;
  count: number;
  options: WireframeAnnotationOptions;
}

export type WireframeAnnotations = WireframeAnnotation[];

type OpenCallback = (isOpen: boolean) => void;

export type WireframeAnnotationAPI = {
  setOptions: (options: APIOptions) => APIOptions;
  getAnnotations: () => WireframeAnnotations;
  register: (Component: ComponentType<any>, options: WireframeAnnotationOptions) => WireframeAnnotation;
  unregister: (Component: ComponentType<any>) => void;
  highlightNote: (Component: ComponentType<any> | undefined) => void;
  setOpen: (isOpen: boolean) => boolean;
  onOpen: (cb: OpenCallback) => {unregister: () => void};
  isOpen: () => boolean;
  setParentReference: (id?: number) => void;
  getParentReference: () => number | undefined;
}

export type withWireframeAnnotationProps = {
  annotationId?: number;
}

export function API(defaultOptions?: APIOptions): WireframeAnnotationAPI {
  let annotations: WireframeAnnotations = [];
  let apiOptions: APIOptions = defaultOptions || {} as APIOptions;
  let parentReference: number | undefined;

  let openCallbacks: OpenCallback[] = [];
  let isOpen = false;

  return {
    setOptions: (options) => {
      apiOptions = options;
      return apiOptions;
    },
    getAnnotations: () => annotations,
    register: (Component, options) => {
      const annotation = getWireframeAnnotation(annotations, Component);

      if (annotation) {
        const updatedAnnotation = {
          ...annotation,
          count: annotation.count + 1,
        };

        annotations = updateWireframeAnnotation(annotations, annotation, updatedAnnotation);

        /* istanbul ignore else */
        if (apiOptions.updater) {
          apiOptions.updater(annotations);
        }

        return updatedAnnotation;
      }

      const newAnnotation = {
        id: annotations.length + 1,
        Component,
        count: 1,
        options,
      };

      annotations = [...annotations, newAnnotation];

      if (apiOptions.updater) {
        apiOptions.updater(annotations);
      }

      return newAnnotation;
    },
    unregister: (Component) => {
      const annotation = getWireframeAnnotation(annotations, Component);

      if (annotation) {
        if (annotation.count > 1) {
          annotations = updateWireframeAnnotation(annotations, annotation, {
            ...annotation,
            count: annotation.count - 1,
          });
        } else {
          annotations = annotations.filter(c => c !== annotation);
        }

        if (apiOptions.updater) {
          apiOptions.updater(annotations);
        }
      }
    },
    highlightNote: Component => apiOptions && apiOptions.highlightNote && apiOptions.highlightNote(getWireframeAnnotation(annotations, Component)),
    setOpen: (open) => {
      if (open !== isOpen) {
        isOpen = open;

        openCallbacks.forEach(cb => cb(isOpen));
      }

      return isOpen;
    },
    onOpen: (cb) => {
      openCallbacks = [...openCallbacks, cb];

      return {
        unregister: () => {
          openCallbacks = openCallbacks.filter(c => c !== cb);
        },
      };
    },
    isOpen: () => isOpen,
    setParentReference: (id) => {
      parentReference = id;
    },
    getParentReference: () => parentReference,
  };
}
