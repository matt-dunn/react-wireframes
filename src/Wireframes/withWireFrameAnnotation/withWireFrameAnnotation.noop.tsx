/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

/* istanbul ignore file */
import { ComponentType } from "react";

export const withWireFrameAnnotation = function withWireFrameAnnotation<P>(WrappedComponent: ComponentType<P> | string) {
  return WrappedComponent;
};
