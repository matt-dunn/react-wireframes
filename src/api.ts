/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import { ComponentType, ReactNode } from "react";
import { remove } from "lodash";

type APIOptions = {
    updater: (components: WireFrameComponents) => void;
    highlightNote?: (component?: WireFrameComponent) => void;
}

export type WireFrameComponentOptions = {
    title: ReactNode;
    description: ReactNode;
}

export type WireFrameComponent = {
    id: number;
    Component: ComponentType<any>;
    count: number;
    options: WireFrameComponentOptions;
}

export type WireFrameComponents = WireFrameComponent[];

type OpenCallback = (isOpen: boolean) => void;

export type WireFrameAnnotationAPI = {
    setOptions: (options: APIOptions) => APIOptions;
    getComponents: () => WireFrameComponents;
    register: (Component: ComponentType<any>, options: WireFrameComponentOptions) => WireFrameComponent;
    unregister: (Component: ComponentType<any>) => void;
    highlightNote: (Component: ComponentType<any> | undefined) => void;
    setOpen: (isOpen: boolean) => boolean;
    onOpen: (cb: OpenCallback) => {unregister: () => void};
    isOpen: () => boolean;
}

export const API = function API(defaultOptions?: APIOptions): WireFrameAnnotationAPI {
  let components: WireFrameComponents = [];
  let apiOptions: APIOptions = defaultOptions || {} as APIOptions;

  let openCallbacks: OpenCallback[] = [];
  let isOpen = false;

  const getComponent = (Component: ComponentType<any> | undefined): WireFrameComponent | undefined => Component && components.find(c => c.Component === Component);

  return {
    setOptions: (options) => {
      apiOptions = options;
      return apiOptions;
    },
    getComponents: () => components,
    register: (Component, options) => {
      const component = getComponent(Component);

      if (component) {
        component.count += 1;

        if (apiOptions.updater) {
          apiOptions.updater(components);
        }

        return component;
      }

      const newComponent = {
        id: components.length + 1,
        Component,
        count: 1,
        options,
      };

      components = [...components, newComponent];

      if (apiOptions.updater) {
        apiOptions.updater(components);
      }

      return newComponent;
    },
    unregister: (Component) => {
      const component = getComponent(Component);

      if (component) {
        component.count -= 1;

        if (component.count === 0) {
          components = remove(components, c => c.Component !== Component);
        }
      }

      if (apiOptions.updater) {
        apiOptions.updater(components);
      }
    },
    highlightNote: Component => apiOptions && apiOptions.highlightNote && apiOptions.highlightNote(getComponent(Component)),
    setOpen: (open) => {
      isOpen = open;
      openCallbacks.forEach(cb => cb(isOpen));
      return isOpen;
    },
    onOpen: (cb) => {
      openCallbacks = [...openCallbacks, cb];
      return {
        unregister: () => {
          openCallbacks = remove(openCallbacks, c => c !== cb);
        },
      };
    },
    isOpen: () => isOpen,
  };
};
