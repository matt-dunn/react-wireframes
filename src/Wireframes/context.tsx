/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React, { createContext, ReactNode } from "react";

import { WireframeAnnotationAPI } from "./api";

export const WireframeAnnotationContext = createContext<WireframeAnnotationAPI | undefined>(undefined);

type WireframeProviderProps = {
  children: ReactNode;
  api: WireframeAnnotationAPI;
}

export const WireframeProvider = ({ children, api }: WireframeProviderProps) => (
  <WireframeAnnotationContext.Provider value={api}>
    {children}
  </WireframeAnnotationContext.Provider>
);
