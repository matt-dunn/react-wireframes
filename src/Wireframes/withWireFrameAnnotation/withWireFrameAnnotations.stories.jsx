import React from "react";
import styled from "@emotion/styled";

import { withWireFrameAnnotation, WireFrameProvider } from "../index";
import { API } from "../api";

export default {
  title: "@matt-dunn/react-wireframes/withWireFrameAnnotation",
  parameters: {
    component: withWireFrameAnnotation,
    componentSubtitle: "Annotate component",
  },
};

const Section = styled.div`
  border-radius: 0.5rem;
  padding: 2rem;
  margin: 0 0 1rem 0;
  background-color: #f0f0f0;
`;

const WASample = withWireFrameAnnotation(Section, {
  title: "Header",
  description: "Header description. Morbi tempor libero id accumsan sodales. Etiam maximus convallis faucibus. Nunc hendrerit sit amet ante in lobortis. Aliquam feugiat nibh sit amet nunc varius laoreet. Aliquam pharetra odio mi, sed convallis massa sagittis at. Nullam nibh tortor, commodo ac risus vitae, venenatis lobortis libero.",
});

const wireframeAPI = API();

export const AnnotatedComponent = () => (
  <WireFrameProvider api={wireframeAPI}>
    <WASample>Header</WASample>
  </WireFrameProvider>
);

const wireframeAPIOpen = API();

wireframeAPIOpen.setOpen(true);

export const AnnotatedComponentOpen = () => (
  <WireFrameProvider api={wireframeAPIOpen}>
    <WASample>Header</WASample>
  </WireFrameProvider>
);
