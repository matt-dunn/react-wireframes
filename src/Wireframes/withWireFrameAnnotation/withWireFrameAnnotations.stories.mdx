import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';

import { API, withWireFrameAnnotation, WireFrameProvider, WireFrameAnnotationProps, withWireFrameAnnotationInterfaceDefinition } from "src/Wireframes";

import { wireframeAPI, wireframeAPIOpen, WASample } from "./withWireFrameAnnotations.stories.fixtures";

<Meta title="@matt-dunn/react-wireframes/withWireFrameAnnotation" component={withWireFrameAnnotation} />

# withWireFrameAnnotation

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi ornare ligula nec leo facilisis ultricies. Fusce quis massa finibus neque volutpat commodo. Nunc blandit dolor in mi viverra, sit amet imperdiet lectus euismod. Mauris et urna vitae dolor accumsan vehicula quis ut urna. Aenean tincidunt enim in pellentesque fringilla. Duis sagittis sem ac quam egestas elementum. Nulla a velit a diam congue euismod.

<Props of={withWireFrameAnnotationInterfaceDefinition} />

<Preview withSource={false}>
    <Story name="Default">
      <WireFrameProvider api={wireframeAPI}>
        <WASample>Header</WASample>
      </WireFrameProvider>
    </Story>
    <Story name="With annotation identifier">
      <WireFrameProvider api={wireframeAPIOpen}>
        <WASample>Header</WASample>
      </WireFrameProvider>
    </Story>
</Preview>


```js
import { withWireFrameAnnotation } from "@matt-dunn/react-wireframes";

const AnnotatedComponent = withWireFrameAnnotation(MyComponent, {
  title: "String title or Component",
  description: "String description or Component",
});
```
