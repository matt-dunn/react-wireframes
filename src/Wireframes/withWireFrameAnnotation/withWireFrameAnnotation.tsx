/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React, {
  ComponentType, ReactNode, useCallback, useContext, useEffect, useState,
} from "react";
import styled from "@emotion/styled";
import css from "@emotion/css";
import { isString } from "lodash";

import { WireFrameComponent, WireFrameComponentOptions } from "../api";
import { WireFrameAnnotationContext } from "../context";

import Identifier from "./Identifier";

type WireFrameAnnotationProps = {
  className?: string;
  children?: ReactNode;
}

const Wrapper = styled.span<{show: boolean}>`
  position: relative;
  
  [disabled] {
    pointer-events: none;
  }
  
  &:hover {
    ${({ show }) => show && css`
      z-index: 5000;
      
      > * {
        box-shadow: 0 0 0 1px #4086f7 !important;
      }
    `}

    > [data-annotation-identifier] {
        transition: opacity 0ms, visibility 0ms;
        opacity: 1;
    }
  }
`;

/* istanbul ignore next */
const getDisplayName = (WrappedComponent: ComponentType<any> | string) => (isString(WrappedComponent) ? WrappedComponent : WrappedComponent.displayName || WrappedComponent.name || "Component");

export function withWireFrameAnnotation<P extends object>(WrappedComponent: ComponentType<P> | string, options: WireFrameComponentOptions) {
  const Component = React.memo<P>((props: P) => <WrappedComponent {...props} />);
  Component.displayName = `withWireFrameAnnotation(${getDisplayName(WrappedComponent)})`;

  function WireFrameAnnotation({ className, ...props }: P & WireFrameAnnotationProps) {
    const api = useContext(WireFrameAnnotationContext);

    if (!api) {
      throw new TypeError("withWireFrameAnnotation does not have the api configured via it's WireFrameProvider");
    }

    const {
      register, unregister, onOpen, isOpen, highlightNote,
    } = api;
    const [annotation, setAnnotation] = useState<WireFrameComponent>();
    const [show, setShow] = useState(isOpen());

    useEffect(() => {
      const cb = onOpen(setShow);
      setAnnotation(register(Component, options));

      return () => {
        cb.unregister();
        unregister(Component);
      };
    }, []); // eslint-disable-line react-hooks/exhaustive-deps

    const handleHighlightNote = useCallback((e) => {
      e.stopPropagation();
      highlightNote(Component);
    }, [highlightNote]);

    const handleHighlightNoteReset = useCallback(() => {
      highlightNote(undefined);
    }, [highlightNote]);

    return (
      <Wrapper
        show={show}
        onMouseOver={handleHighlightNote}
        onFocus={handleHighlightNote}
        onMouseLeave={handleHighlightNoteReset}
        className={className}
      >
        {annotation && <Identifier annotation={annotation} show={show} />}

        <Component {...props as P} />
      </Wrapper>
    );
  }

  WireFrameAnnotation.Component = Component;

  return WireFrameAnnotation;
}
