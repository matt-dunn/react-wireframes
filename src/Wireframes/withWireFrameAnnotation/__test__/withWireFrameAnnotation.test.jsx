/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React from "react";
import { mount, shallow } from "enzyme";
import { act } from "react-dom/test-utils";

import { WireFrameProvider } from "../../context";
import { API } from "../../api";

import { withWireFrameAnnotation } from "../withWireFrameAnnotation";

describe("Wireframe: withWireFrameAnnotation", () => {
  let api;
  let Fragment;
  let WrappedComponent;
  let ComponentTree;
  let highlightNote;

  beforeEach(() => {
    highlightNote = jest.fn();

    api = API({
      highlightNote,
    });

    WrappedComponent = withWireFrameAnnotation(() => <div>MOCK COMPONENT</div>, {
      title: <div>Title</div>,
      description: <div>Description.</div>,
    });

    ComponentTree = <WrappedComponent />;

    Fragment = (
      <WireFrameProvider api={api}>
        {ComponentTree}
      </WireFrameProvider>
    );
  });

  it("should render without identifier when closed", () => {
    const wrapper = mount(Fragment);

    const annotation = api.getAnnotations()[0];

    expect(annotation).toEqual({
      id: 1,
      count: 1,
      Component: WrappedComponent.Component,
      options: {
        title: <div>Title</div>,
        description: <div>Description.</div>,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("should render annotation identifier when open", () => {
    const wrapper = mount(Fragment);

    act(() => {
      api.setOpen(true);
    });

    expect(highlightNote).not.toHaveBeenCalled();

    expect(wrapper).toMatchSnapshot();
  });

  it("should highlight annotation", () => {
    const wrapper = mount(Fragment);

    act(() => {
      api.setOpen(true);
    });

    expect(highlightNote).not.toHaveBeenCalled();

    wrapper.simulate("mouseover");

    expect(highlightNote).toHaveBeenCalledWith(api.getAnnotations()[0]);

    wrapper.update();

    expect(wrapper).toMatchSnapshot();

    wrapper.simulate("mouseleave");

    expect(highlightNote).toHaveBeenCalledWith(undefined);
  });

  it("should unregister when unmounted", () => {
    const wrapper = mount(Fragment);

    expect(api.getAnnotations().length).toEqual(1);

    wrapper.unmount();
    expect(api.getAnnotations().length).toEqual(0);
  });

  it("should throw exception if api is not provided", () => {
    expect(() => {
      shallow(ComponentTree);
    }).toThrow(TypeError);
  });
});
