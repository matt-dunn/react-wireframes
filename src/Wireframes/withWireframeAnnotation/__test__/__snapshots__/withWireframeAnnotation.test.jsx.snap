// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wireframe: withWireframeAnnotation should highlight annotation 1`] = `
.emotion-7 {
  position: relative;
}

.emotion-7 [disabled] {
  pointer-events: none;
}

.emotion-7:hover {
  z-index: 5000;
}

.emotion-7:hover > * {
  box-shadow: 0 0 0 1px #4086f7 !important;
}

.emotion-7:hover > [data-annotation-identifier] {
  -webkit-transition: opacity 0ms,visibility 0ms;
  transition: opacity 0ms,visibility 0ms;
  opacity: 1;
}

.emotion-3 {
  position: absolute;
  top: -1em;
  left: -1em;
  z-index: 4000;
  cursor: default;
  -webkit-transition: opacity 250ms,visibility 250ms;
  transition: opacity 250ms,visibility 250ms;
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.enter {
  opacity: 0;
  visibility: hidden;
}

.emotion-3.fade-Identifier.enter-active {
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.exit {
  opacity: 0.75;
  visibility: visible;
}

.emotion-3.fade-Identifier.exit-active {
  opacity: 0;
  visibility: hidden;
}

.emotion-1 {
  border-radius: 2em;
  background-color: yellow;
  border-color: #caca00;
  white-space: nowrap;
  min-width: 2em;
  height: 2em;
  padding: 0.25em 0.5em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-style: normal;
  font-weight: normal;
  position: absolute;
  top: -1em;
  left: -1em;
  z-index: 4000;
  cursor: default;
  -webkit-transition: opacity 250ms,visibility 250ms;
  transition: opacity 250ms,visibility 250ms;
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.enter {
  opacity: 0;
  visibility: hidden;
}

.emotion-1.fade-Identifier.enter-active {
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.exit {
  opacity: 0.75;
  visibility: visible;
}

.emotion-1.fade-Identifier.exit-active {
  opacity: 0;
  visibility: hidden;
}

<WireframeProvider
  api={
    Object {
      "getAnnotations": [Function],
      "highlightNote": [Function],
      "isOpen": [Function],
      "onOpen": [Function],
      "register": [Function],
      "setOpen": [Function],
      "setOptions": [Function],
      "unregister": [Function],
    }
  }
>
  <WrappedWireframeAnnotation>
    <Wrapper
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      show={true}
    >
      <span
        className="emotion-7 emotion-8"
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <IdentifierComponent
          annotation={
            Object {
              "Component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "displayName": "withWireframeAnnotation(Component)",
                "type": [Function],
              },
              "count": 1,
              "id": 1,
              "options": Object {
                "description": <div>
                  Description.
                </div>,
                "title": <div>
                  Title
                </div>,
              },
            }
          }
          show={true}
        >
          <CSSTransition
            className="fade-Identifier"
            classNames=""
            in={true}
            mountOnEnter={true}
            timeout={250}
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="fade-Identifier"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={250}
              unmountOnExit={true}
            >
              <IdentifierContainer
                annotation={
                  Object {
                    "Component": Object {
                      "$$typeof": Symbol(react.memo),
                      "compare": null,
                      "displayName": "withWireframeAnnotation(Component)",
                      "type": [Function],
                    },
                    "count": 1,
                    "id": 1,
                    "options": Object {
                      "description": <div>
                        Description.
                      </div>,
                      "title": <div>
                        Title
                      </div>,
                    },
                  }
                }
                className="fade-Identifier"
                data-annotation-identifier={true}
              >
                <Identifier
                  annotation={
                    Object {
                      "Component": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "displayName": "withWireframeAnnotation(Component)",
                        "type": [Function],
                      },
                      "count": 1,
                      "id": 1,
                      "options": Object {
                        "description": <div>
                          Description.
                        </div>,
                        "title": <div>
                          Title
                        </div>,
                      },
                    }
                  }
                  className="fade-Identifier emotion-3 emotion-0"
                  data-annotation-identifier={true}
                >
                  <IdentifierContainer
                    className="fade-Identifier emotion-3 emotion-0"
                    data-annotation-identifier={true}
                  >
                    <cite
                      className="fade-Identifier emotion-0 emotion-1 emotion-2"
                      data-annotation-identifier={true}
                    >
                      1
                    </cite>
                  </IdentifierContainer>
                </Identifier>
              </IdentifierContainer>
            </Transition>
          </CSSTransition>
        </IdentifierComponent>
        <withWireframeAnnotation(Component)>
          <Component>
            <div>
              MOCK COMPONENT
            </div>
          </Component>
        </withWireframeAnnotation(Component)>
      </span>
    </Wrapper>
  </WrappedWireframeAnnotation>
</WireframeProvider>
`;

exports[`Wireframe: withWireframeAnnotation should render annotation identifier when open 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-0 [disabled] {
  pointer-events: none;
}

.emotion-0:hover > [data-annotation-identifier] {
  -webkit-transition: opacity 0ms,visibility 0ms;
  transition: opacity 0ms,visibility 0ms;
  opacity: 1;
}

<WireframeProvider
  api={
    Object {
      "getAnnotations": [Function],
      "highlightNote": [Function],
      "isOpen": [Function],
      "onOpen": [Function],
      "register": [Function],
      "setOpen": [Function],
      "setOptions": [Function],
      "unregister": [Function],
    }
  }
>
  <WrappedWireframeAnnotation>
    <Wrapper
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      show={false}
    >
      <span
        className="emotion-0 emotion-1"
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <IdentifierComponent
          annotation={
            Object {
              "Component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "displayName": "withWireframeAnnotation(Component)",
                "type": [Function],
              },
              "count": 1,
              "id": 1,
              "options": Object {
                "description": <div>
                  Description.
                </div>,
                "title": <div>
                  Title
                </div>,
              },
            }
          }
          show={false}
        >
          <CSSTransition
            className="fade-Identifier"
            classNames=""
            in={false}
            mountOnEnter={true}
            timeout={250}
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="fade-Identifier"
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={250}
              unmountOnExit={true}
            />
          </CSSTransition>
        </IdentifierComponent>
        <withWireframeAnnotation(Component)>
          <Component>
            <div>
              MOCK COMPONENT
            </div>
          </Component>
        </withWireframeAnnotation(Component)>
      </span>
    </Wrapper>
  </WrappedWireframeAnnotation>
</WireframeProvider>
`;

exports[`Wireframe: withWireframeAnnotation should render without identifier when closed 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-0 [disabled] {
  pointer-events: none;
}

.emotion-0:hover > [data-annotation-identifier] {
  -webkit-transition: opacity 0ms,visibility 0ms;
  transition: opacity 0ms,visibility 0ms;
  opacity: 1;
}

<WireframeProvider
  api={
    Object {
      "getAnnotations": [Function],
      "highlightNote": [Function],
      "isOpen": [Function],
      "onOpen": [Function],
      "register": [Function],
      "setOpen": [Function],
      "setOptions": [Function],
      "unregister": [Function],
    }
  }
>
  <WrappedWireframeAnnotation>
    <Wrapper
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      show={false}
    >
      <span
        className="emotion-0 emotion-1"
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <IdentifierComponent
          annotation={
            Object {
              "Component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "displayName": "withWireframeAnnotation(Component)",
                "type": [Function],
              },
              "count": 1,
              "id": 1,
              "options": Object {
                "description": <div>
                  Description.
                </div>,
                "title": <div>
                  Title
                </div>,
              },
            }
          }
          show={false}
        >
          <CSSTransition
            className="fade-Identifier"
            classNames=""
            in={false}
            mountOnEnter={true}
            timeout={250}
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="fade-Identifier"
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={250}
              unmountOnExit={true}
            />
          </CSSTransition>
        </IdentifierComponent>
        <withWireframeAnnotation(Component)>
          <Component>
            <div>
              MOCK COMPONENT
            </div>
          </Component>
        </withWireframeAnnotation(Component)>
      </span>
    </Wrapper>
  </WrappedWireframeAnnotation>
</WireframeProvider>
`;
